pkg/controllermanager/controller/event/event_control_test.go

```diff
@@ -0,0 +1,208 @@
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("#reconcileEvent", func() {
+	var (
+		ctrl                   *gomock.Controller
+		clientMap              *fakeclientmap.ClientMap
+		k8sGardenRuntimeClient *mockclient.MockClient
+
+		namespaceName = "garden-foo"
+		requestName   = "request"
+
+		ttl = &metav1.Duration{Duration: 1 * time.Hour}
+
+		eventController            *Controller
+		request                    reconcile.Request
+		shootEvent                 *corev1.Event
+		nonShootEvent              *corev1.Event
+		nonGardenerAPIGroupEvent   *corev1.Event
+		eventWithoutInvolvedObject *corev1.Event
+		cfg                        *config.EventControllerConfiguration
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		k8sGardenRuntimeClient = mockclient.NewMockClient(ctrl)
+		clientMap = fakeclientmap.NewClientMap().AddRuntimeClient(keys.ForGarden(), k8sGardenRuntimeClient)
+
+		logger.Logger = logger.NewNopLogger()
+
+		shootEvent = &corev1.Event{
+			LastTimestamp:  metav1.Time{Time: time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+			InvolvedObject: corev1.ObjectReference{Kind: "Shoot", APIVersion: "core.gardener.cloud/v1beta1"},
+		}
+		nonShootEvent = &corev1.Event{
+			LastTimestamp:  metav1.Time{Time: time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+			InvolvedObject: corev1.ObjectReference{Kind: "Project", APIVersion: "core.gardener.cloud/v1beta1"},
+		}
+		eventWithoutInvolvedObject = &corev1.Event{
+			LastTimestamp: metav1.Time{Time: time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+		}
+		nonGardenerAPIGroupEvent = &corev1.Event{
+			LastTimestamp:  metav1.Time{Time: time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+			InvolvedObject: corev1.ObjectReference{Kind: "Shoot", APIVersion: "v1"},
+		}
+		request = reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: namespaceName,
+				Name:      requestName,
+			}}
+		cfg = &config.EventControllerConfiguration{
+			TTLNonShootEvents: ttl,
+		}
+
+		eventController = NewController(clientMap, cfg)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Context("Shoot Events", func() {
+
+		It("should ignore them", func() {
+			mockClientGet(k8sGardenRuntimeClient, request.NamespacedName, shootEvent)
+			expected := reconcile.Result{}
+			actual, err := eventController.reconcileEvent(request)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actual).To(Equal(expected))
+		})
+	})
+
+	Context("Non-Shoot events", func() {
+
+		Context("ttl is not yet reached", func() {
+
+			BeforeEach(func() {
+				nowFunc = func() time.Time {
+					return time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)
+				}
```

Comment:
Can you ensure, that `nowFunc` is reset after each test? As you are mocking it for every case in the moment, it doesn't really matter, but it is best practice to reset faked/mocked variables after tests to avoid executing incorrect tests when new cases are added in the future.

You can use our helper func `test.WithVar[s]` for that (see [here](https://github.com/gardener/gardener/blob/056f925a09162fb90fcce938b04c0397250bf25f/pkg/operation/botanist/extensions/infrastructure/infrastructure_test.go#L107-L110) for an example).